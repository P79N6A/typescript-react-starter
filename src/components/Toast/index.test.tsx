import * as React from 'react';
import { shallow } from 'enzyme';
import { Toast } from './index';
import * as renderer from 'react-test-renderer';

describe('<Toast />', () => {
  // shallow render with enzyme
  it('renders without crashing', () => {
    shallow(<Toast text="hello world" />);
  });

  // Just teast enzyme function
  it('text', () => {
    const text = 'hello world';
    const wrapper = shallow(<Toast text={text} />);
    expect(wrapper.text()).toBe(text);
    expect(wrapper.contains(text)).toBe(true);
  });

  // Snapshot Testing with Jest
  it('renders correctly', () => {
    const tree = renderer.create(<Toast text="hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
