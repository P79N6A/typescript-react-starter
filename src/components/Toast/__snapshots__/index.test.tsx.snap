// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> renders correctly 1`] = `
<div
  className="toast-container"
>
  hello world
</div>
`;
